### PR TITLE
Upgrade to .NET 8

### PR DESCRIPTION
**Pull request by danielchalmers**
- Updated packages
- Declared target framework in Dependency.Build.props
- Most importantly updated the cache from `blazor-resources-/` to `dotnet-resources-/`.

### CODE REVIEW

**Comment by mikes-gh** [comment]
Looking promising.
New set of eyes has worked well here 🙏 
I'll ask @ScarletKuro to test the bind scenario he mentioned on discord.

File: src/MudBlazor.Examples.Data/MudBlazor.Examples.Data.csproj [hunk 1]
```diff
@@ -1,7 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net7.0</TargetFramework>
   </PropertyGroup>
```
Comment (mikes-gh) [inline src/MudBlazor.Examples.Data/MudBlazor.Examples.Data.csproj:4]: I guess you can remove the outer `<PropertyGroup>` tags?

File: src/Try.Core/Try.Core.csproj [hunk 2]
```diff
@@ -1,15 +1,14 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net7.0</TargetFramework>
   </PropertyGroup>
```
Comment (mikes-gh) [inline src/Try.Core/Try.Core.csproj:4]: I guess you can remove the outer `<PropertyGroup>` tags?

**Review by mikes-gh — COMMENTED** [review]
Some small alterations

**Comment by mikes-gh** [comment]
Thanks a lot @danielchalmers 
If you want you can do the refactoring to net8 `program.cs` new methods once this is merged

**Comment by danielchalmers** [comment]
> If you want you can do the refactoring to net8 `program.cs` new methods once this is merged

I'll leave it for now because it is "Just Working" (tm)

**Pull request merged**